xopt/generators/bayesian/models/standard.py

```diff
@@ -10,20 +10,21 @@
 from gpytorch.kernels import Kernel
 from gpytorch.likelihoods import GaussianLikelihood
 from gpytorch.priors import GammaPrior
-from pydantic import Field, validator
+from pydantic import ConfigDict, Field, field_validator
 from torch.nn import Module
 
 from xopt.generators.bayesian.base_model import ModelConstructor
 from xopt.generators.bayesian.models.prior_mean import CustomMean
 from xopt.generators.bayesian.utils import get_input_transform, get_training_data
-from xopt.pydantic import orjson_dumps
+from xopt.pydantic import decode_torch_module
+from pydantic_core.core_schema import FieldValidationInfo
 
 DECODERS = {"torch.float32": torch.float32, "torch.float64": torch.float64}
 MIN_INFERRED_NOISE_LEVEL = 1e-4
 
 
 class StandardModelConstructor(ModelConstructor):
-    name = "standard"
+    name: str = Field("standard", frozen=True)
```

Comment:
this should probably be a class variable instead, if you could update this and the base constructor class that would be great

Reply:
although on second thought, maybe not since the name isn't dumped during serialization. What do you think?

Reply:
Yes, classvars not getting dumped was the issue so kept them as standard fields. In case of generator, where name is classvar, we manually add that key during dump. There is another option of having a hidden discriminator field but that can't be (directly) used because they are incompatible with pre-validators.